### ISSUE TITLE
Table column expansion (`*`) doesn't work with select from function

### ISSUE THREAD

**Issue by KES777** [dbeaver/dbeaver#12623]
**Is your feature request related to a problem? Please describe.**
it is boring to type all column names.

**Describe the solution you'd like**
at many places editor do proposals. 
count `table.**` as proposal `table.col1, table.col2, table.col3`

details:
https://www.postgresql.org/message-id/668748662.20210522113754%40yandex.ru

**Comment by ShadelessFox** [comment]
Hello @KES777,

You can expand table columns by pressing <kbd>Ctrl+Space</kbd> with cursor placed just after `*`. 
Consider following examples with cursor denoted by `|` symbol:

```sql
SELECT *| FROM Table;
---
SELECT column1, column2, ... FROM Table;
```
```sql
SELECT Table.*| FROM Table;
---
SELECT Table.column1, Table.column2, ... FROM Table;
```
```sql
SELECT t.*| FROM Table t;
---
SELECT t.column1, t.column2, ... FROM Table;
```

Let me know if you still have any questions left.

**Comment by KES777** [comment]
![image](https://user-images.githubusercontent.com/799021/119967509-16f3e300-bfb5-11eb-89dc-2afe3fdd2ac1.png)

expanding does not work for functions. It just put FROM when I use hotkey. There is not any spacing before keyword, which is syntax error

**Comment by ShadelessFox** [comment]
We don't support that. I will rename the ticket to reflect the actual problem.

**Comment by ShadelessFox** [comment]
I'm concerned about this improvement.
In order to expand the columns we need to obtain them first. In case of functions, we need to execute it to get the result. It can cause problems and damage if function is not pure (changes data) and the user is unaware of this behavior.

**Comment by devscheffer** [comment]
It also doesn't work when you try with WITH() like 
```
with base as (select 1 as c1 from table1)
select b.* from base b
```
If you try ctrl+space won't bring the right column c1

**Comment by p91paul** [comment]
@ShadelessFox I expect most databases have metadata about the function return type that can be queried without executing it?

**Comment by ShadelessFox** [comment]
@p91paul, they do, but I suspect in some databases a function may have a dynamic return type. In that case we can't do nothing. 

That's a nice point anyway!

**Comment by p91paul** [comment]
@ShadelessFox of course, but I also think that with a dynamic return type the user cannot have a reasonable expectation of getting autocompletion!
